### PR TITLE
[Documentation] Route Manager Design doc

### PR DESCRIPTION
**Pull request by cj-chung**
This design document includes concept, data schemas, APIs, and detailed workflows design for the management of routers, routing tables and routing rules in Alcor for both Neutron and VPC scenarios.

### CODE REVIEW

File: docs/modules/ROOT/pages/mgmt_services/route_manager.adoc [hunk 1]
```diff
@@ -0,0 +1,1227 @@
+= Route Manager Design Specification
+Chun-Jen Chung <cchung@futurewei.com>, Liguang Xie <lxie@futurewei.com>
+v0.1, 2020-08-17
+:toc: right
+:imagesdir: ../../images
+
+== Overview
+
+The Route Manager is an Alcor microservice which manages the route tables and route entries for all VPCs and Subnets.
+It also provides router-related functionalities to compatible with OpenStack Neutron's router APIs.
```
Comment (xieus) [inline docs/modules/ROOT/pages/mgmt_services/route_manager.adoc:10]: Minor: to be compatible with

File: docs/modules/ROOT/pages/mgmt_services/route_manager.adoc [hunk 2]
```diff
@@ -0,0 +1,1227 @@
...
+
+[arabic]
+. Availability Zone and Subnet-based route table.
+. Default route tables for VPC and subnet.
+. Default route to Internet Gateway in a public subnet route table.
+. Default route to NAT Gateway in a private subnet route table.
+. Router resource managed by an owner, Neutron or VPC.
+. Route table resource managed by an owner, VPC or Subnet.
+. Route entry should include destination, nexthop, target, and priority.
+. Warning message should be raised if same route entry with different priority has been inserted by users.
```
Comment (xieus) [inline docs/modules/ROOT/pages/mgmt_services/route_manager.adoc:25]: One more requirement:
- Warning message should be raised when adding an invalid routing rule

File: docs/modules/ROOT/pages/index.adoc [hunk 3]
```diff
@@ -13,7 +13,7 @@
 *** xref:mgmt_services/vpc_manager.adoc[VPC Manager]
```
Comment (xieus) [inline docs/modules/ROOT/pages/index.adoc:13]: We will need a new route_manager_l3_neighbors_update_5.PNG to remove the grammar check warning.

**Review by xieus — COMMENTED** [review]
Look great in general. One comment is to continue on designing the Subnet Mgr path, and keep APIs up to update.

File: docs/modules/ROOT/pages/mgmt_services/route_manager.adoc (hunk 1 above)
Comment (cj-chung) [reply docs/modules/ROOT/pages/mgmt_services/route_manager.adoc]: updated

File: docs/modules/ROOT/pages/mgmt_services/route_manager.adoc (hunk 2 above)
Comment (cj-chung) [reply docs/modules/ROOT/pages/mgmt_services/route_manager.adoc]: updated

File: docs/modules/ROOT/pages/index.adoc (hunk 3 above)
Comment (cj-chung) [reply docs/modules/ROOT/pages/index.adoc:13]: The PNG is recaptured without grammar check warning.

**Review by xieus — APPROVED** [review]
LGTM. Thanks @cj-chung for putting all the design details together.

**Pull request merged**